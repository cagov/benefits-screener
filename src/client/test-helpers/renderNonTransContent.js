import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
