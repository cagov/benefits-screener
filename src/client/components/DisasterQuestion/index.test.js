import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<DisasterQuestion />", () => {
  it("renders the component", async () => {
    const wrapper = renderNonTransContent(Component, {
      onChange: () => {},
    });

    expect(wrapper).toMatchSnapshot();
  });
});
