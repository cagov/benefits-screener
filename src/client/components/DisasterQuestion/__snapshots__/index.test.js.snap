// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisasterQuestion /> renders the component 1`] = `
<FormGroup
  className="follow-up disaster-question unchecked"
  controlId="disaster-question"
>
  <fieldset>
    <FormLabel
      as="legend"
      column={false}
      srOnly={false}
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou have been diagnosed with COVID-19."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You have been diagnosed with COVID-19."
      label="You have been diagnosed with COVID-19."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-1"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou have COVID-19 symptoms and are seeking a diagnosis."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You have COVID-19 symptoms and are seeking a diagnosis."
      label="You have COVID-19 symptoms and are seeking a diagnosis."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-2"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioA member of your household has COVID-19."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="A member of your household has COVID-19."
      label="A member of your household has COVID-19."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-3"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou are taking care of your family or household member who has COVID-19."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You are taking care of your family or household member who has COVID-19."
      label="You are taking care of your family or household member who has COVID-19."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-4"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou had to quit your job as a direct result of COVID-19."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You had to quit your job as a direct result of COVID-19."
      label="You had to quit your job as a direct result of COVID-19."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-5"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYour place of employment is closed as a direct result of COVID-19. (This may apply if your business is open or partially re-opened but your position is not needed due to COVID-19 limitations on the business)."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="Your place of employment is closed as a direct result of COVID-19. (This may apply if your business is open or partially re-opened but your position is not needed due to COVID-19 limitations on the business)."
      label="Your place of employment is closed as a direct result of COVID-19. (This may apply if your business is open or partially re-opened but your position is not needed due to COVID-19 limitations on the business)."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-6"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou cannot reach your workplace because of a quarantine as a direct result of COVID-19. (This may apply if you cannot reach your workplace because of a state or municipal order restricting travel to combat the spread of COVID-19, such as a stay-home or shelter-in-place order.)"
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You cannot reach your workplace because of a quarantine as a direct result of COVID-19. (This may apply if you cannot reach your workplace because of a state or municipal order restricting travel to combat the spread of COVID-19, such as a stay-home or shelter-in-place order.)"
      label="You cannot reach your workplace because of a quarantine as a direct result of COVID-19. (This may apply if you cannot reach your workplace because of a state or municipal order restricting travel to combat the spread of COVID-19, such as a stay-home or shelter-in-place order.)"
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-7"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou cannot reach your workplace because your health care provider advised you to quarantine due to COVID-19 concerns."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You cannot reach your workplace because your health care provider advised you to quarantine due to COVID-19 concerns."
      label="You cannot reach your workplace because your health care provider advised you to quarantine due to COVID-19 concerns."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-8"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou cannot work because you have primary responsibility for caring for a child or another person and their school or care facility is closed as a direct result of COVID-19."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You cannot work because you have primary responsibility for caring for a child or another person and their school or care facility is closed as a direct result of COVID-19."
      label="You cannot work because you have primary responsibility for caring for a child or another person and their school or care facility is closed as a direct result of COVID-19."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-9"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou had a definite date to start a job that is no longer available as a direct result of COVID-19."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You had a definite date to start a job that is no longer available as a direct result of COVID-19."
      label="You had a definite date to start a job that is no longer available as a direct result of COVID-19."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-10"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou had a definite date to start a job but cannot reach that job as a direct result of COVID-19."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You had a definite date to start a job but cannot reach that job as a direct result of COVID-19."
      label="You had a definite date to start a job but cannot reach that job as a direct result of COVID-19."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-11"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou are an independent contractor with reportable income (ex. IRS Form 1099) and you are forced to stop working because COVID-19 has severely limited your ability to continue performing your customary work activities."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You are an independent contractor with reportable income (ex. IRS Form 1099) and you are forced to stop working because COVID-19 has severely limited your ability to continue performing your customary work activities."
      label="You are an independent contractor with reportable income (ex. IRS Form 1099) and you are forced to stop working because COVID-19 has severely limited your ability to continue performing your customary work activities."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-12"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioYou became the major financial support for your household because the head of household died as a direct result of COVID-19."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="You became the major financial support for your household because the head of household died as a direct result of COVID-19."
      label="You became the major financial support for your household because the head of household died as a direct result of COVID-19."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-13"
    />
    <FormCheck
      checked={false}
      disabled={false}
      id="disaster-question-radioNone of these options apply to me."
      inline={true}
      isInvalid={false}
      isValid={false}
      key="None of these options apply to me."
      label="None of these options apply to me."
      name="disaster-question"
      onChange={[Function]}
      required={true}
      title=""
      type="radio"
      value="choice-14"
    />
    <Feedback
      type="invalid"
    >
      This field is required.
    </Feedback>
  </fieldset>
</FormGroup>
`;
